src/dnf-plugins/product-id/test-product-id.c

```diff
@@ -93,11 +94,21 @@ typedef struct {
 } handleFixture;
 
 void setup(handleFixture *fixture, gconstpointer testData) {
+    GError *error = NULL;
     (void)testData;
     fixture->dnfContext = dnf_context_new();
+    dnf_context_setup(fixture->dnfContext, NULL, &error);
     PluginMode mode = PLUGIN_MODE_CONTEXT;
-    // This is never explicitly called
-    fixture->handle = pluginInitHandle(1, mode, (void*)fixture->dnfContext);
+    // When plugin product-id is called by libdnf application, then pluginInitHandle is called,
+    // but this function pluginInitHandle should not be never called by other application
+    // directly, because only libdnf can set DNF context properly. We have to set attributes
+    // of handle using following way.
+    // fixture->handle = pluginInitHandle(1, mode, (void*)fixture->dnfContext);
```

Comment:
leaving this here is a bit confusing, as it seems either a leftover, or something that was intended to be like this

Reply:
Changed comment and added more explanation.